timemachine/cpp/src/context.cu

```diff
@@ -192,9 +192,14 @@ std::array<std::vector<double>, 2> Context::multiple_steps_local(
 
         int num_row_indices = p_num_selected.data[0];
         int num_col_indices = N_ - num_row_indices;
-        if (num_row_indices == 0 || num_col_indices == 0) {
-            throw std::runtime_error(
-                "entire system selected; reduce radius, increase k, check system or use Context::multiple_steps");
+
+        if (num_row_indices == 0) {
+            throw std::runtime_error("Context::multiple_steps_local(): no free particles selected");
```

Comment:
Related to this [previous discussion](https://github.com/proteneer/timemachine/pull/975#discussion_r1134732982), wondering whether there's any concern about this being triggered in long simulations (even for reasonable parameter values, there is still some small probability, correct?). Would it make sense for this to be converted to a warning, similarly to the "entire system selected" case?

Reply:
At the moment this crashes if the exception isn't raise as the code currently tries to run a kernel with 0 blocks. Though  we could make it work if this seems like a case to handle

Reply:
Would also need to change NonbondedAllPairs (https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/nonbonded_all_pairs.cu#L153-L155)

We could also do something like the following to avoid running any GPU code for this case
```
if (num_row_indices == 0) {
    copy_current_x_and_box_to_output_buffer()
    return std::array<std::vector<double>, 2>({h_x_buffer, h_box_buffer});
}
```

Reply:
> Makes sense to raise an error if this is not currently a case we handle. Would it make sense to raise the error from the site where we make an assumption about the number of blocks being nonzero (neighborlist?). That way it'll be obvious what needs to be changed if this ever does become an issue.

Since this is currently coming out of a kernel launch failure, don't think we want it to fail from the site, or if we do we would probably want an exception and not a call to exit. Already seems like multiple layers to the failure (kernel launch, NonbondedAllPairs, probably another) so the digging will have to occur regardless

Reply:
This also makes it clear that we hit a case we don't support, rather than potentially hitting a bug due to an unexpected case.

Reply:
I think it's fine as-is for now. Seems potentially useful to raise exceptions nearer to where assumptions are made in addition, looking ahead to when we want to enable this and remove the check in this PR. But I don't think it's necessary for this PR.

Reply:
Will look at `radius >= reasonable_radius and k > reasonable_k`, initial thought is `reasonable_radius = 0.1` and `reasonable_k = 1`

Reply:
Made changes to enforce 'reasonable' values here [354dfbe](https://github.com/proteneer/timemachine/pull/981/commits/354dfbedf02cebaf800a36fffc1e6c845e0b20df)

Had to change test to be on pathological case. Should be easy to use to make this case work in the future if we need it.